planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_planner_module.cpp

```diff
@@ -1354,19 +1367,34 @@ BehaviorModuleOutput GoalPlannerModule::planPullOverAsOutput(
     // update goal candidates
     auto goal_candidates = thread_safe_data_.get_goal_candidates();
     goal_searcher_->update(
-      goal_candidates, occupancy_grid_map_, planner_data_,
-      thread_safe_data_.get_static_target_objects());
+      goal_candidates, occupancy_grid_map_, planner_data_, context_data.static_target_objects);
 
     // Select a path that is as safe as possible and has a high priority.
-    const auto pull_over_path_candidates = thread_safe_data_.get_pull_over_path_candidates();
+    const auto & pull_over_path_candidates = context_data.pull_over_path_candidates;
     const auto path_and_goal_opt =
       selectPullOverPath(context_data, pull_over_path_candidates, goal_candidates);
 
     // update thread_safe_data_
     if (path_and_goal_opt) {
       auto [pull_over_path, modified_goal] = *path_and_goal_opt;
-      deceleratePath(pull_over_path);
+      /** TODO(soblin): since thread_safe_data::pull_over_path was used as a global variable, old
+       * code was setting deceleration to thread_safe_data::pull_over_path and setOutput() accessed
+       * to the velocity profile in thread_safe_data::pull_over_path, which is a very bad usage of
+       * member variable
+       *
+       * set this selected pull_over_path to ThreadSafeData, but actually RoadParking thread does
```

Comment:
minor question: road parking is equal to lane parking?

Reply:
Yes